app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -69,14 +78,14 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
 
         ObjectAnimator animation = ObjectAnimator.ofFloat(binding.pop, "rotation", 0f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f);
         animation.setDuration(1000);
-        binding.pop.setOnClickListener(view1 -> UIHelpers.darkModeToggle(v, animation, this.getContext()));
+        binding.pop.setOnClickListener(view1 -> UIHelpers.darkModeToggle(v, animation, this.getContext(), () -> binding.location.setTextColor(Color.parseColor((MainActivity.scoutLocation < 3 ? "#FF0000" : "#0000FF")))));
```

Comment:
I recommend assigning these literal values to constants. It's generally not  a good idea to just sprinkle magic numbers in your code.